library/headers.inc.php

```diff
@@ -42,43 +46,66 @@ function include_css_library($path)
 */
 function call_required_libraries($library_array){
-/* First checking for any library in a directory matching the string "/jquery-min-/".
- * When one is found, use that value in the URL string, then add "index.js"
- */
-    foreach($library_array as $v){ 
-    if (preg_match("/jquery-min-/",$v)) {?>
-    <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path'].$v ; ?>/index.js"></script>    
-    <?php 
-    } }
+    /* First checking for any library in a directory matching the string "/jquery-min-/".
+     * When one is found, use that value in the URL string, then add "index.js"
+     * In order to decrease the chance of including and library calling problems, some measures were taken:
+     * - The array in the argument is completely converted to lower-case, meaning that there is not any sensitivity between jQuery & jquery or fancyBox & fancybox
+     * - Some libraries can be called with or without "-". For example: "font-awesome" and "fontawesome" => those 2 are the same thing
+     */
+    
+    $library_array = array_map('strtolower', $library_array);
+
+    foreach ($library_array as $v){ 
+        if (preg_match("/jquery-min-/", $v)) {?>
+            <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path'].$v ; ?>/index.js"></script>    
+        <?php 
+        }
+    }   
         
-    if (in_array("bootstrap",$library_array)){   ?>
+    if (in_array("bootstrap", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['standard_js_path']; ?>bootstrap-3-3-4/dist/css/bootstrap.min.css" type="text/css">
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>bootstrap-3-3-4/dist/js/bootstrap.min.js"></script>
     <?php
     }
-    if (in_array("fancybox",$library_array)){   ?>
+
+    if (in_array("fancybox", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>fancybox/jquery.fancybox-1.2.6.css" media="screen" />
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>fancybox/jquery.fancybox-1.2.6.js"></script>
     <?php
     }
-    if (in_array("knockout",$library_array)){   ?>
+
+    if (in_array("knockout", $library_array)) {   ?>
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>knockout/knockout-3.4.0.js"></script>
     <?php
     }
-    if (in_array("datepicker",$library_array)){   ?>
+
+    if (in_array("datepicker", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>jquery-datetimepicker/jquery.datetimepicker.css" media="screen" />
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>jquery-datetimepicker/jquery.datetimepicker.full.min.js"></script>
     <?php
     }
-    if (in_array('font-awesome', $library_array)) {?>
+
+    if (in_array('font-awesome', $library_array) || in_array('fontawesome', $library_array)) {   ?>
```

Comment:
Do we have an existing inconsistency here?

Reply:
Didn't get it. Are you talking about the font-awesome thing? I've explained it in the comments after call_required_libraries

Calling **fontawesome** is the same as calling **font-awesome**

Reply:
So you are fearing someone using a wrong argument in the function, that is right?  OK I can live with that.  In that case, using a capital Font-Awesome or the like will also break it on Linux.  I don't necessarily have a problem with this, but I don't feel it is necessary.

Reply:
Yes, my fear was that 😋 
When calling a library, sometimes we need to check at headers.inc.php to know exactly what is the library name to put into the argument. But I can remove this if you say so